zmanim/astronomical_calendar.py

```diff
@@ -0,0 +1,103 @@
+from zmanim.util.noaa_calculator import NOAACalculator
+from zmanim.util.geo_location import GeoLocation
+from datetime import date, datetime, timedelta
+import pytz
+from zmanim.util.astronomical_calculations import AstronomicalCalculations
+from zmanim.util.math_helper import MathHelper
+
+
+class AstronomicalCalendar(MathHelper):
+    GEOMETRIC_ZENITH = 90
+    CIVIL_ZENITH = 96
+    NAUTICAL_ZENITH = 102
+    ASTRONOMICAL_ZENITH = 108
+
+    __sentinel=object()
+
+    def __init__(self, geo_location: GeoLocation = None, date: date = None, calculator: AstronomicalCalculations = None):
```

Comment:
Since the arguments are being assigned None default values, the arg type for those is technically `Optional[<type>]`.  
This could be left as is, since mypy is smart enough to deduce the type is Optional here, but it's good to know that according to PEP 484, `Type checkers should move towards requiring the optional type to be made explicit.`
https://www.python.org/dev/peps/pep-0484/#union-types

Reply:
might be getting nitpicky on the types -- mypy can be configured to be more or less strict, so really just depends.

Reply:
don't think being careful about types is being nit-picky. Modules with good typehints are tremendously valueable, especially when you use them with a good IDE like PyCharm, etc.

Reply:
To enforce the explicit optional, you can run mypy with the `--no-implicit-optional` flag.